bin/github-linguist

```diff
@@ -8,106 +8,142 @@ require 'json'
 require 'optparse'
 require 'pathname'
 
-path = ARGV[0] || Dir.pwd
+HELP_TEXT = <<~HELP
+Linguist v#{Linguist::VERSION}
+Detect language type and determine language breakdown for a given Git repository.
 
-# special case if not given a directory
-# but still given the --breakdown or --json options/
-if path == "--breakdown"
-  path = Dir.pwd
-  breakdown = true
-elsif path == "--json"
+Usage: linguist <path>
+      linguist <path> [--breakdown] [--json]
+      linguist [--breakdown] [--json]
+HELP
+
+def github_linguist(args)
+  breakdown = false
+  json_output = false
   path = Dir.pwd
-  json_breakdown = true
-end
 
-ARGV.shift
-breakdown = true if ARGV[0] == "--breakdown"
-json_breakdown = true if ARGV[0] == "--json"
+  parser = OptionParser.new do |opts|
+    opts.banner = HELP_TEXT
+
+    opts.on("-b", "--breakdown", "Breakdown stuff") { breakdown = true }
+    opts.on("-j", "--json", "Output results in JSON format") { json_output = true }
+    opts.on("-h", "--help", "Prints this help") do
+      puts opts
+      exit
+    end
+  end
```

Comment:
Nit: indentation...

```suggestion
    opts.banner = HELP_TEXT

    opts.on("-b", "--breakdown", "Breakdown stuff") { breakdown = true }
    opts.on("-j", "--json", "Output results in JSON format") { json_output = true }
    opts.on("-h", "--help", "Prints this help") do
    puts opts
    exit
  end
end
```

Reply:
I don't think this change makes sense. The lines:
```ruby
put opts
exit
```
are part of a `do` block for the `--help` option, which is why they're indented the way they are. Also, the last `end` in this code block is closing out the `OptionsParser.new` block, not the surrounding method.

```ruby
parser = OptionsParser.new do |opts|
  ...
  opts.on("-h", "--help", "Prints this help") do
    puts opts
    exit
  end
end
```

Reply:
🤦 oh yes. I didn't see it was part of `do `blocks.